provider/workspace_preset.go

```diff
@@ -12,59 +12,82 @@ import (
 type WorkspacePreset struct {
 	Name       string            `mapstructure:"name"`
 	Parameters map[string]string `mapstructure:"parameters"`
+	Prebuilds  WorkspacePrebuild `mapstructure:"prebuilds"`
+}
+
+type WorkspacePrebuild struct {
+	Instances int `mapstructure:"instances"`
 }
 
 func workspacePresetDataSource() *schema.Resource {
 	return &schema.Resource{
 		SchemaVersion: 1,
 
-		Description: "Use this data source to predefine common configurations for workspaces.",
+		Description: "Use this data source to predefine common configurations for coder workspaces. Users will have the option to select a defined preset, which will automatically apply the selected configuration. Any parameters defined in the preset will be applied to the workspace. Parameters that are not defined by the preset will still be configurable when creating a workspace.",
 		ReadContext: func(ctx context.Context, rd *schema.ResourceData, i interface{}) diag.Diagnostics {
 			var preset WorkspacePreset
```

Comment:
@SasSwart btw: I just noticed code in the `ReadContext` is basically no-op?
It reads data and then forgets them.
Probably it's fine if main goal it to send this data downstream to `provisioner`

But still what is the purpose of this code?
- is it for validation purposes? But I'm not sure maybe validations happens earlier?
- is it for helping debugging in the future?

Reply:
Most of the decoding is defunct, yes. I've removed it.
I don't think we should rely on this for validation.
We define a schema below, which Terraform already uses for validation.
I've removed the defunct code here: https://github.com/coder/terraform-provider-coder/pull/376